actions/templates/put/index.js

```diff
@@ -0,0 +1,197 @@
+const Enforcer = require('openapi-enforcer');
+const consoleLib = require('@adobe/aio-lib-console');
+
+const HTTP_METHOD = 'put';
+const PUT_PARAM_NAME = 'templateId';
+
+const serializeRequestBody = (params) => {
+  return {
+    ...(params.description && { description: params.description }), // developer console only
+    ...(params.latestVersion && { latestVersion: params.latestVersion }), // developer console only
+    ...(params.updatedBy && { updatedBy: params.updatedBy }),
+    ...('adobeRecommended' in params && { adobeRecommended: params.adobeRecommended }),
+    ...('keywords' in params && params.keywords.length && { keywords: params.keywords }),
+    ...('categories' in params && params.categories.length && { categories: params.categories }),
+    ...('extensions' in params && params.extensions.length && { extensions: params.extensions }),
+    ...('credentials' in params && params.credentials.length && { credentials: params.credentials }),
+    ...('codeSamples' in params && params.codeSamples.length && { codeSamples: params.codeSamples }),
+    ...('apis' in params && params.apis.length && { apis: params.apis }),
+    ...('status' in params && { status: params.status }),
+    ...('runtime' in params && { runtime: params.runtime }),
+    ...('links' in params && { links: params.links })
+  };
+};
+
+/**
+ * Updates a new template in the Template Registry.
+ * @param {object} params request parameters
+ * @returns {object} response
+ */
+async function main (params) {
+  // create a Logger
+  const logger = Core.Logger('main', { level: params.LOG_LEVEL || 'info' });
+
+  const imsUrl = params.IMS_URL;
+  const imsClientId = params.IMS_CLIENT_ID;
+
+  const dbParams = {
+    MONGODB_URI: params.MONGODB_URI,
+    MONGODB_NAME: params.MONGODB_NAME
+  };
+
+  try {
+    // 'info' is the default level if not set
+    logger.info('Calling "PUT templates"');
+
+    // log parameters, only if params.LOG_LEVEL === 'debug'
+    logger.debug(stringParameters(params));
+
+    if (params.__ow_method === undefined || params.__ow_method.toLowerCase() !== HTTP_METHOD) {
+      return errorResponse(405, [errorMessage(ERR_RC_HTTP_METHOD_NOT_ALLOWED, `HTTP "${params.__ow_method}" method is unsupported.`)], logger);
+    }
+
+    // check for missing request input parameters and headers
+    const requiredHeaders = ['Authorization'];
+    const errorMessages = checkMissingRequestInputs(params, [], requiredHeaders);
+    if (errorMessages) {
+      return errorResponse(401, errorMessages, logger);
+    }
+
+    const isTemplateIdValid = PUT_PARAM_NAME in params && typeof params[PUT_PARAM_NAME] === 'string' && params[PUT_PARAM_NAME].length > 0;
+    if (!isTemplateIdValid) {
+      return errorResponse(400, [errorMessage(ERR_RC_MISSING_REQUIRED_PARAMETER, `The "${PUT_PARAM_NAME}" parameter is not set.`)], logger);
+    }
+
+    // extract the user Bearer token from the Authorization header
+    const accessToken = getBearerToken(params);
```

Comment:
access token validation is done at the runtime level now based on our swagger security definition, no need to validate again here